### PR TITLE
Staff Resource Restrictions

### PR DESCRIPTION
**Pull request by markottaviani**
Implements #404 

Ready to Merge

### CODE REVIEW

**Comment by markottaviani** [comment]
@TheDudeWithTheThing planning on merging and releasing this tomorrow if you have any thoughts on it.

**Comment by TheDudeWithTheThing** [comment]
sorry buddy!  looking now

File: spec/controllers/folders_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      before :each do
+        allow(controller).to receive(:current_user) { admin }
+      end
+      it 'is able to view files in standard folders' do
+        unlocked_folder = create :folder
+        get :show, id: unlocked_folder
+        expect(assigns(:folder)).to eq unlocked_folder
+      end
+      it 'is able to view files in restricted folders' do
+        locked_folder = create(:folder, locked: TRUE)
```
Comment (TheDudeWithTheThing) [inline spec/controllers/folders_controller_spec.rb]: you could turn these into traits in the factory so you can do

`create(:folder, :locked)`
`create(:folder, :unlocked)`

File: spec/factories/folder.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FactoryGirl.define do
+  factory :folder do
+    name 'Folder Name'
+    description 'Folder Descritpion'
+  end
+end
```
Comment (TheDudeWithTheThing) [inline spec/factories/folder.rb]: :no_entry_sign: 

:(

**Comment by TheDudeWithTheThing** [comment]
nothing blocking, looks good!

**Pull request merged**